### PR TITLE
BS-13480 Studio help text in Property sections wrapping bevahiour

### PR DESCRIPTION
**Pull request by rbioteau**
Fix wraping layout for section help
Implement a Well widget to increase help visibility

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import org.eclipse.swt.events.PaintEvent;
+import org.eclipse.swt.events.PaintListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.forms.IFormColors;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+
+public class Well extends Composite {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: Why is it called Well?

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private final Label label;
+
+    public Well(final Composite parent, final String text, final FormToolkit toolkit, final int severity) {
+        super(parent, SWT.NONE);
+        setLayout(GridLayoutFactory.fillDefaults().extendedMargins(10, 10, 5, 8).create());
+
+        addPaintListener(new PaintListener() {
+
+            @Override
+            public void paintControl(final PaintEvent e) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: playing with paint, we need to test it on Linux and mac

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                return warningSeprator;
+            case IStatus.ERROR:
+                return errorSeprator;
+            default:
+                break;
+        }
+        return toolkit.getColors().getColor(IFormColors.SEPARATOR);
+    }
+
+    private Color backgroundrColor(final FormToolkit toolkit, final int style) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: typo backGroundrColor --> backGroundColor

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java (hunk 1 above)
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: CF ergo corner

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java (hunk 2 above)
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: Yes.

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.PaintEvent;
+import org.eclipse.swt.events.PaintListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+
+public class Well extends Composite {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: why is it called Well?

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: This is how a such widget is named (cf ergo corner)

Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: ok seen in Ergo Corner, but Google seems to not know it :s maybe we should add some javadoc to explain what it is

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/properties/Well.java]: Javadoc added

**Comment by rbioteau** [comment]
Rebased

**Pull request merged**